### ISSUE TITLE
Cogserver and attention circular dependencies?

### ISSUE THREAD

**Issue by eddiemonroe** [opencog/cogserver#8]
Trying to build cogserver, I get these errors:
```
In file included from /home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:47:0:
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:104:3: error: ‘AVCHSigl’ does not name a type
   AVCHSigl* _avchange_signal;
   ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:107:3: error: ‘AVCHSigl’ does not name a type
   AVCHSigl* _add_af_signal;
   ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:110:3: error: ‘AVCHSigl’ does not name a type
   AVCHSigl* _remove_af_signal;
   ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:134:24: error: ‘AttentionValuePtr’ has not been declared
   Json::Value avToJSON(AttentionValuePtr av);
                        ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:160:30: error: ‘AttentionValuePtr’ does not name a type
                        const AttentionValuePtr& av_old,
                              ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:161:30: error: ‘AttentionValuePtr’ does not name a type
                        const AttentionValuePtr& av_new);
                              ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:166:26: error: ‘AttentionValuePtr’ does not name a type
                    const AttentionValuePtr& av_old,
                          ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:167:26: error: ‘AttentionValuePtr’ does not name a type
                    const AttentionValuePtr& av_new);
                          ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:169:29: error: ‘AttentionValuePtr’ does not name a type
                       const AttentionValuePtr& av_old,
                             ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.h:170:29: error: ‘AttentionValuePtr’ does not name a type
                       const AttentionValuePtr& av_new);
                             ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc: In member function ‘void opencog::AtomSpacePublisherModule::enableSignals()’:
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:122:1: error: ‘ttentionValuePtr’ was not declared in this scope
 ttentionValuePtr& av_old,
 ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:122:19: error: ‘av_old’ was not declared in this scope
 ttentionValuePtr& av_old,
                   ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:123:48: error: expected primary-expression before ‘const’
                                                const AttentionValuePtr& av_new)
                                                ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:132:47: error: qualified-id in declaration before ‘(’ token
 void AtomSpacePublisherModule::TVChangedSignal(const Handle& h,
                                               ^
/home/eddie/hansonrobotics/OpenCog/cogserver/opencog/cogserver/modules/events/AtomSpacePublisherModule.cc:339:1: error: expected ‘}’ at end of input
 }
 ^
```
It looks to me the errors involve unrecognized types that are defined in the attention package--AttentionValuePtr and AVCHSigl
```
error: ‘AttentionValuePtr’ does not name a type
```
But cogserver is a prerequisite for the attention package--attention can't be built unless cogserver is already built
```
  attention/build$ cmake ..
  . . .
  CMake Error at CMakeLists.txt:180 (MESSAGE):
  CogServer missing: it is needed!
```
Is there a circular dependency here?

**Comment by linas** [comment]
Yikes! It would appear so. Do you actually need the event publisher, or is it getting built by default?  

One (quasi-unrelated) problem is that the even publisher has bit-rotted; it doesn't publish value change events, etc. and given that it's unmaintained, the easiest fix would be to just stub it out so that it doesn't build.  Is that a viable option?

**Comment by eddiemonroe** [comment]
Nope, I don't need the event publisher. Just need attention and cogserver
as prerequisites to opencog.

**Comment by linas** [comment]
I just merged #9 to stub this out. It should be enough to fix this for you; open a new issue if it still doesn't work...

**Comment by eddiemonroe** [comment]
That fixed it. Thanks, Linas.
